front/components/assistant_builder/ActionScreen.tsx

```diff
@@ -364,91 +385,91 @@ export default function ActionScreen({
-                        label={SEARCH_MODE_SPECIFICATIONS[key].label}
-                        icon={SEARCH_MODE_SPECIFICATIONS[key].icon}
-                        description={
-                          SEARCH_MODE_SPECIFICATIONS[key].description
-                        }
-                        onClick={() => {
-                          setEdited(true);
-                          setBuilderState((state) => ({
-                            ...state,
-                            actionMode:
-                              SEARCH_MODE_SPECIFICATIONS[key].actionMode,
-                          }));
-                        }}
-                      />
-                    ))}
+                    }).map((key) => {
+                      const spec = SEARCH_MODE_SPECIFICATIONS[key];
+                      const defaultAction = getDefaultActionConfiguration(
+                        spec.actionType
+                      );
+                      return (
+                        <DropdownMenu.Item
+                          key={key}
+                          label={spec.label}
+                          icon={spec.icon}
+                          description={spec.description}
+                          onClick={() => {
+                            setEdited(true);
+                            setBuilderState((state) => {
+                              const newBuilderState: AssistantBuilderState = {
+                                ...state,
+                                actions: removeNulls([defaultAction]),
+                              };
+                              return newBuilderState;
+                            });
+                          }}
+                        />
+                      );
+                    })}
                   </DropdownMenu.Items>
                 </DropdownMenu>
               </div>
             )}
           </>
         )}
 
-        <ActionModeSection show={builderState.actionMode === "GENERIC"}>
+        <ActionModeSection show={!action}>
           <div className="pb-16"></div>
         </ActionModeSection>
 
         <ActionModeSection
-          show={
-            builderState.actionMode === "RETRIEVAL_SEARCH" && !noDataSources
-          }
+          show={action?.type === "RETRIEVAL_SEARCH" && !noDataSources}
         >
           <ActionRetrievalSearch
             owner={owner}
-            builderState={builderState}
+            action={action}
             dataSources={dataSources}
             setBuilderState={setBuilderState}
             setEdited={setEdited}
           />
         </ActionModeSection>
 
         <ActionModeSection
-          show={
-            builderState.actionMode === "RETRIEVAL_EXHAUSTIVE" && !noDataSources
-          }
+          show={action?.type === "RETRIEVAL_EXHAUSTIVE" && !noDataSources}
         >
           <ActionRetrievalExhaustive
             owner={owner}
-            builderState={builderState}
+            action={action}
```

Comment:
can we ensure we pass the right type here? with

`{ action?.type === "RETRIEVAL_EXHAUSTIVE" && (...) }`

 ?

Reply:
Yeah we can but we kind of have to check downstream anyway for type-safety reasons. Do you feel it's worth-it ?

Reply:
Ok I passed the config directly